SDK v2 migration guide.md

```diff
@@ -0,0 +1,235 @@
+# SDK v2 migration guide
+
+This document outlines the changes made from this library's 1.X.X releases to its 2.X.X releases. Since this is
+a major version upgrade, there are a number of breaking changes that will affect the ability to compile. Provided here
+are outlines of the notable breaking changes as well as a mapping from v1 APIs to v2 APIs to aid migrating.
+
+## Table of contents
+
+ - [Why the v1 SDK is being replaced](#Why-the-v1-sdk-is-being-replaced)
+ - [What will happen to the v1 SDK](#What-will-happen-to-the-v1-sdk)
+ - [Migration guide](#migration-guide)
+   - [IoT hub device client](#iot-hub-device-client)
+   - [IoT hub service client](#iot-hub-service-client)
+   - [Device Provisioning Service (DPS) device client](#dps-device-client)
+   - [DPS Service Client](#dps-service-client)
+   - [Security provider clients](#security-provider-clients)
+   - [Deps](#deps)
+ - [Frequently asked questions](#frequently-asked-questions)
+
+## Why the v1 SDK is being replaced
+
+There are a number of reasons why the Azure IoT SDK team chose to do a major version revision. Here are a few of the more important reasons:
+  - [Upcoming certificate changes](./upcoming_certificate_changes_readme.md) dictated that the SDK needed to stop pinning on a specific IoT Hub public certificate and start reading certificates from the device certificate store.
+  - Removing or upgrading several NuGet dependencies (TODO: list).
+  - Consolidate IoT hub service clients and rename to reflect the items or operations they support. Many existing client classes (RegistryManager, ServiceClient, etc.) were confusingly named and contained methods that weren't always consistent with the client's assumed responsibilities.
+  - Many existing clients had a mix of standard constructors (```new DeviceClient(...)```) and static builder methods (```DeviceClient.CreateFromConnectionString(...)```) that caused some confusion among users. The factory methods have been removed and the addition of constructors in clients enables unit testing.
```

Comment:
```suggestion
  - Many existing clients had a mix of standard constructors (`new DeviceClient(...)`) and static builder methods (`DeviceClient.CreateFromConnectionString(...)`) that caused some confusion among users. The factory methods have been removed and the addition of constructors in clients enables unit testing.
```

Reply:
I don't know why there are triple back ticks. They aren't needed to show as "code".